include/sound/soc-acpi.h

```diff
@@ -87,6 +88,8 @@ struct snd_soc_acpi_mach_params {
 	unsigned short subsystem_vendor;
 	unsigned short subsystem_device;
 	bool subsystem_id_set;
+	enum snd_soc_acpi_intel_codec codec_type;
+	enum snd_soc_acpi_intel_codec amp_type;
```

Comment:
One could argue that this could be done in the machine driver.

The only argument to detect the codec_type and amp_type in the parent is that this could potentially avoid duplication of code in machine drivers, but you haven't really described this in the commit message.

I would revisit the commit message to first say what the issue is and why it makes sense to add new parameters.

Oh and you also need to update the kernel-doc after line 75 to add the new parameters.